tests/test_bookmarks_updated.py

```diff
@@ -133,27 +140,33 @@ def bookmarks_test(self, conn_id, testable_streams):
-                    # verify the 2nd sync bookmark value is the max replication key value for a given stream
-                    self.assertLessEqual(replication_key_value, second_bookmark_value_utc, msg="Second sync bookmark was set incorrectly, a record with a greater replication key value was synced")
+                # The `transactions` stream is a child of the `orders` stream. Hence the bookmark for transactions is solely dependent on the value of bookmark in 'transaction_orders' which stores the parent record's bookmark.
+                # Hence it doesn't have its own replication key.
+                if stream not in ('transactions'):
+                    replication_key = next(iter(expected_replication_keys[stream]))
+                    for record in first_sync_messages:
+                        replication_key_value = record.get(replication_key)
+                        # verify 1st sync bookmark value is the max replication key value for a given stream
+                        self.assertLessEqual(replication_key_value, first_bookmark_value_utc, msg="First sync bookmark was set incorrectly, a record with a greater replication key value was synced")
+
+                    for record in second_sync_messages:
+                        replication_key_value = record.get(replication_key)
+                        # verify the 2nd sync replication key value is greater or equal to the 1st sync bookmarks
+                        self.assertGreaterEqual(replication_key_value, simulated_bookmark_value, msg="Second sync records do not respect the previous                                                  bookmark")
+                        # verify the 2nd sync bookmark value is the max replication key value for a given stream
+                        self.assertLessEqual(replication_key_value, second_bookmark_value_utc, msg="Second sync bookmark was set incorrectly, a record with a greater replication key value was synced")
 
                 # verify that we get less data in the 2nd sync
                 # collects has all the records with the same value of replication key, so we are removing from this assertion
-                if stream not in ('collects'):
+                # As the bookmark for `transactions` is solely dependent on the value of bookmark in 'transaction_orders' which stores the parent record's
+                # bookmark, hence we'd now get all the data for transactions stream without filtering on `created_at`
+                if stream not in ('collects', 'transactions'):
                     self.assertLess(second_sync_count, first_sync_count,
                                     msg="Second sync does not have less records, bookmark usage not verified")
 
                 # verify that we get atleast 1 record in the second sync
-                if stream not in ('collects'):
+                if stream not in ('collects', 'transactions'):
```

Comment:
If there is no data replicated for these streams on the 2nd sync, then I don't think there is much benefit to having them in this test. They should be marked as not under test and removed from `expected_streams`. The `expected_streams` or some variable like that should drive the whole test including table selection, assertions, etc. That way adding/removing streams to the test in the future is just a matter of updating a set or list in this test file. It should be done for each test not in base.py.

Reply:
As suggested in the earlier comment, we are now discarding the transactions bookmark. However, we do have data for the stream. We did this particular change to skip the assertion for the stream.

Reply:
My bad, we had data fro transactions, hence added back the stream for that particular assertion